rpc/client.go

```diff
@@ -90,12 +90,9 @@ type ClientInterface interface {
 type Client struct {
 	sync.RWMutex
 
-	upstreamEnabled bool
-	upstreamURL     string
 	UpstreamChainID uint64
```

Comment:
Keep the UpstreamChainID because it's accessed from several places:
accounts.go, call_raw.go, personal/api.go, rpcfilters/api.go and so on.

Reply:
yeap, I feel like all those other pieces of code using this can be removed as well. AFAIU we've got a bunch of code in place to ensure that status-go exposes the standard ETH JSON-RPC interface. Are we aware of any status-go client using these capabilities?

Reply:
![image](https://github.com/user-attachments/assets/a1baf638-e7c1-41cb-a352-7abd85f01726)
I see 7 years old code, so I feel there isn't any :D

Reply:
Some methods, like `AddCustomToken` in `call_raw.go`, aren't even called from anywhere, and there are 25 occurances to check, so investigating will take some time.
I'm unsure about some of these spots. If it's okay with you, I can clean this up and create a separate ticket for it, as some of the code doesn't seem outdated

Reply:
https://github.com/status-im/status-go/issues/5937